Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCreateTransmission.py

```diff
@@ -0,0 +1,179 @@
+            doc="A list of input run numbers. Multiple runs will be summed after loading",
+        )
+
+        self.copyProperties(self._TRANS_WS_ALG, [Prop.TRANS_ROI, Prop.I0_MON_IDX, Prop.MON_WAV_MIN, Prop.MON_WAV_MAX])
+
+        self.declareProperty(
+            MatrixWorkspaceProperty(Prop.FLOOD_WS, "", direction=Direction.Input, optional=PropertyMode.Optional),
+            doc="The workspace to be used for the flood correction. If this property is not set then no flood correction is performed.",
+        )
+
+        self.declareProperty(
+            Prop.BACK_SUB_ROI,
+            "",
+            doc=f"The set of workspace indices to be passed to the ProcessingInstructions property of the {self._BACK_SUB_ALG} algorithm."
+            " If this property is not set then no background subtraction is performed.",
+        )
+
+        self.declareProperty(
+            WorkspaceProperty(Prop.OUTPUT_WS, "", direction=Direction.Output),
+            doc="The output transmission workspace",
+        )
+
+    def PyExec(self):
+        ws = self._load_and_sum_runs(self.getProperty(Prop.INPUT_RUNS).value, self.getPropertyValue(Prop.OUTPUT_WS))
+
+        ws = self._apply_flood_correction(ws)
+        ws = self._apply_background_subtraction(ws)
+        ws = self._create_transmission_ws(ws)
+
+        self.setProperty(Prop.OUTPUT_WS, ws)
+
+    def _load_and_sum_runs(self, runs: list[str], output_name: str):
+        """Load and sum the input runs"""
+        self.log().information("Loading and summing the run files")
+        alg = self.createChildAlgorithm(self._LOAD_ALG, Filename="+".join(runs), LoaderName="LoadNexus", OutputWorkspace=output_name)
+        alg.setRethrows(True)
+        alg.setProperty("MergeRunsOptions", {"FailBehaviour": "Stop"})
+        alg.execute()
+        return alg.getProperty("OutputWorkspace").value
+
+    def _apply_flood_correction(self, workspace):
+        """If a flood workspace has been provided then apply a flood correction"""
+        flood_ws = self.getProperty(Prop.FLOOD_WS).value
+        if flood_ws is None:
+            return workspace
+
+        self.log().information("Performing flood correction")
+        args = {"FloodWorkspace": flood_ws}
+        corrected_ws = self._run_algorithm(workspace, self._FLOOD_ALG, "InputWorkspace", args)
+
+        # The flood correction performs a divide, which can result in the workspace being set as a distribution
+        # if the flood and input workspaces have the same Y units. We need to set the resulting workspace back to
+        # a non-distribution as it may be dimensionless after the divide, but it shouldn't be a distribution.
+        if isinstance(corrected_ws, WorkspaceGroup):
+            for child_ws in corrected_ws:
+                child_ws.setDistribution(False)
+        else:
+            corrected_ws.setDistribution(False)
+
+        return corrected_ws
+
+    def _apply_background_subtraction(self, workspace):
```

Comment:
The algorithm `CreateTransmissionWorkspace` is also subtracting a flat background. What's the difference here? Can this be omitted?

Reply:
It's an interesting question, but I don't know whether you would get the same result if you took this out - you don't seem to be able to pass a background region to subtract in the way you can with `ReflectometryBackgroundSubtraction`, for example? I haven't looked too closely at it as the task with this one was to implement the same workflow as POLREF use in their own code, and they use `ReflectometryBackgroundSubtraction` followed by `CreateTransmissionWorkspace`, so I stuck with that.